REFACTOR - HISTORY - Search conversions by money

Deivid117 opened the pull request:
- Se cambió el código de búsqueda de conversiones para poder buscar también por el nombre de la columna (pesosBob, pesosArg y Dollar), por lo que se implementó un RawQuery para poder hacer consultas en tiempo de ejecución ya que el nombre de la columna cambia dinámicamente según lo que seleccione el usuario.

[Tareas](https://trello.com/c/gUcLPoli)

Deivid117 committed:
REFACTOR - Raw query to search conversions

- Se actualizó el query de la búsqueda de conversiones por el raw query para poder modificar el query de búsqueda en tiempo de ejecución
- Se agregó el parámetro de columnName para el repository, impl y use case para las posibles búsquedas por columna(pesosBob, pesosArg, dollar)

Deivid117 committed:
REFACTOR - Query updated

- Se quitó de la búsqueda de las conversiones el parámetro de la fecha, ahora solo puede buscar por el nombre y la divisa que selecciones (Bob, Arg, Dollar)

Deivid117 committed:
Merge branch 'develop' of https://github.com/VarquLabs/DolarBlueApp into refactor/search-conversions-by-money

# Conflicts:
#	app/src/main/java/com/varqulabs/dolarblue/history/data/local/database/dao/ConversionsHistoryDao.kt
#	app/src/main/java/com/varqulabs/dolarblue/history/data/repository/ConversionsHistoryRepositoryImpl.kt
#	app/src/main/java/com/varqulabs/dolarblue/history/domain/repository/ConversionsHistoryRepository.kt

Deivid117 committed:
REFACTOR - Get grouped conversions for search conversions

- Se utilizó el método que agrupa las conversiones para buscar conversiones y reutilizar código

Deivid117 committed:
REFACTOR - remove id

- Se quitó el id del método de agrupación ya que no se requiere

juanjosevarqu commented on app/src/main/java/com/varqulabs/dolarblue/history/data/local/database/dao/ConversionsHistoryDao.kt:
```diff
@@ -32,13 +34,6 @@ interface ConversionsHistoryDao {
-     """)
-    fun searchConversionsHistoryByQuery(querySearch: String): Flow<List<ConversionsWithCurrentExchangeRelation>>
+    @RawQuery
+    fun searchConversionsHistoryByQuery(query: SupportSQLiteQuery): Flow<List<ConversionsWithCurrentExchangeRelation>>
```
Seria bueno que nos regales una breve explicacion de cuando conviene usar los RawQuery

Deivid117 replied:
va

juanjosevarqu commented on app/src/main/java/com/varqulabs/dolarblue/history/domain/useCases/SearchConversionsHistoryUseCase.kt:
```diff
@@ -7,7 +7,7 @@ import kotlinx.coroutines.flow.Flow
 ) {
-    suspend operator fun invoke(searchQuery: String): Flow<List<ConversionsHistory>> {
-        return repository.searchConversionsHistoryByQuery(searchQuery)
+    suspend operator fun invoke(columnName: String, searchQuery: String): Flow<List<ConversionsHistory>> {
```
Para invocar al Caso de Uso de manera segura lo haremos mediante Constantes ?

Deivid117 replied:
¿A qué te refieres con constantes?

juanjosevarqu reviewed (commented):
Deje un par de dudas

Deivid117 committed:
REFACTOR - ConversionSearch model

- Se creó un modelo para con los atributos necesarios para la búsqueda de conversiones.
- Se renombró la función de búsqueda de conversiones.

Deivid117 committed:
Merge branch 'develop' of https://github.com/VarquLabs/DolarBlueApp into refactor/search-conversions-by-money

Deivid117 committed:
HOTFIX - Model name changed

- Se cambió el nombre del modelo que se usa para buscar las conversiones en el repository

juanjosevarqu reviewed (approved):
cracken
